extensions/amp-inline-gallery/0.1/amp-inline-gallery-thumbnails.js

```diff
@@ -0,0 +1,241 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {CarouselEvents} from '../../amp-base-carousel/0.1/carousel-events';
+import {InlineGalleryEvents} from './inline-gallery-events';
+import {createCustomEvent} from '../../../src/event-helper';
+import {dict} from '../../../src/utils/object';
+import {htmlFor} from '../../../src/static-template';
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {matches, scopedQuerySelector} from '../../../src/dom';
+import {setStyle} from '../../../src/style';
+
+/**
+ * Renders a carousel of thumbnails for an inline gallery.
+ *
+ * TODO(sparhami) Look into syncing the state between the main carousel and
+ * the thumbnails. One potential solution is to only change the position when
+ * the new slide would be outside of the currently visibile thumbnails. More
+ * investigation / UX work is needed here.
+ * TODO(sparhami) Change amp-base-carousel to move one viewport of items at
+ * a time when using `mixed-length="true"`.
+ * TODO(sparhami) Look into styling for the active thumbnail.
+ * TODO(sparhami) Make non-looping thumbnails the default; need to make sure
+ * the next arrow works properly for that case.
+ */
+export class AmpInlineGalleryThumbnails extends AMP.BaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?Element} */
+    this.carousel_ = null;
+
+    /** @private {number} */
+    this.thumbAspectRatio_ = 0;
```

Comment:
I think it's probably cleaner to make this `null` since 0 can still be interpreted as an aspect ratio

Reply:
Done